lib/widgets/highlight_text.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+
+class HighlightText extends StatelessWidget {
+  final String text;
+  final String? searchWord;
+  final TextStyle? style;
+  final TextStyle? highlightStyle;
+
+  const HighlightText({
+    super.key,
+    required this.text,
+    required this.searchWord,
+    required this.style,
+    required this.highlightStyle,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    if ((searchWord?.isEmpty ?? true) || text.isEmpty) {
+      return Text(text, style: style);
+    }
+
+    final List<TextSpan> spans = [];
+    text.splitMapJoin(
+      RegExp(searchWord!, caseSensitive: false),
+      onMatch: (Match match) {
+        spans.add(buildHighlightSpan(match.group(0)!));
+        return '';
+      },
+      onNonMatch: (String nonMatch) {
+        spans.add(buildNormalSpan(nonMatch));
+        return '';
+      },
+    );

```

Comment:
I think you should put if in an extension, or at least in a method

Reply:
done in 007a3586e85c17686fbfee3c87e4f2a010a9435e